crypto/x509/x509_vfy.c

```diff
@@ -146,8 +148,6 @@ static X509 *lookup_cert_match(X509_STORE_CTX *ctx, X509 *x)
             break;
         xtmp = NULL;
     }
-    if (xtmp != NULL && !X509_up_ref(xtmp))
-        xtmp = NULL;
```

Comment:
I've changed this is because this PR uses `lookup_cert_match()` also directly in `build_chain() ` in a way where there is no need to keep a reference to the result.
Also the existing use in `check_trust()` becomes simpler if the `up_ref()` is done there in case really needed, rather than freeing the reference again in case it turns out it is not actually needed.
So in essence this is an internal code cleanup.

Reply:
I don't presently know why the latter change is needed. We should probably figure that out first.

Reply:
Yes, although, as mentioned, the proposed move of `up_ref()` to where really needed is also more generally a code simplification.

Reply:
This change is wrong. It is not safe to remove the up_ref, while still returning the certificate, which may be freed before the function returns by the line below, which is needed to free the rest of the list.

Reply:
Well, if `ctx->lookup_certs` is `lookup_certs_sk()` or `X509_STORE_CTX_get1_certs()`, then the reference counter of the returned certs is certainly >1, so the cert returned, `xtmp`, cannot actually be freed prematurely by `sk_X509_pop_free(certs, X509_free)`.
But still good point, there is the risk that a function set via `X509_STORE_set_lookup_certs()` does not guarantee this.
So reverted this change.